web_src/js/components/ScopedAccessTokenSelector.vue

```diff
@@ -0,0 +1,156 @@
+    <div class="field gt-pl-2">
+      <label class="checkbox-label">
+        <input
+          ref="category"
+          v-model="categorySelected"
+          class="scope-checkbox scoped-access-token-input"
+          type="checkbox"
+          name="scope"
+          :value="'delete:' + category"
+          @change="onCategoryInput"
+        >
+        {{ category }}
+      </label>
+    </div>
+    <div class="field gt-pl-4">
+      <div class="inline field">
+        <label class="checkbox-label">
+          <input
+            ref="read"
+            v-model="readSelected"
+            :disabled="categorySelected || writeSelected"
+            class="scope-checkbox scoped-access-token-input"
+            type="checkbox"
+            name="scope"
+            :value="'read:' + category"
+            @change="onIndividualInput"
+          >
+          read:{{ category }}
+        </label>
+      </div>
+      <div class="inline field">
+        <label class="checkbox-label">
+          <input
+            ref="write"
+            v-model="writeSelected"
+            :disabled="categorySelected"
+            class="scope-checkbox scoped-access-token-input"
+            type="checkbox"
+            name="scope"
+            :value="'write:' + category"
+            @change="onIndividualInput"
+          >
+          write:{{ category }}
+        </label>
+      </div>
+      <div class="inline field">
+        <label class="checkbox-label">
+          <input
+            ref="delete"
+            v-model="deleteSelected"
+            :disabled="categorySelected"
+            class="scope-checkbox scoped-access-token-input"
+            type="checkbox"
+            name="scope"
+            :value="'delete:' + category"
+            @change="onIndividualInput"
+          >
+          delete:{{ category }}
+        </label>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+
+import {createApp} from 'vue';
+
+const sfc = {
+  props: {
+    category: {
+      type: String,
+      required: true,
+    },
+  },
+
+  data: () => ({
+    categorySelected: false,
+    readSelected: false,
+    writeSelected: false,
+    deleteSelected: false,
+  }),
+
+  methods: {
+    /**
+     * When entire category is toggled
+     * @param {Event} event
+     */
+    onCategoryInput(event) {
+      event.preventDefault();
+      this.deleteSelected = this.$refs.category.checked;
+      this.writeSelected = this.$refs.category.checked;
+      this.readSelected = this.$refs.category.checked;
+    },
+
+    /**
+     * When an individual level of category is toggled
+     * @param {Event} event
+     */
+    onIndividualInput(event) {
+      event.preventDefault();
+      if (this.$refs.delete.checked) {
+        this.readSelected = true;
+        this.writeSelected = true;
+        this.categorySelected = true;
+      }
+      if (this.$refs.write.checked) {
+        this.readSelected = true;
+      }
+    },
+  }
```

Comment:
I've noticed a slight disconnect between the behaviors which I think we should fix:
When you click on a top level checkbox, all sub-checkboxes are checked.
By clicking on the checkbox again, all sub-checkboxes are deselected.
Okay, so far so good.

However, if you click on the `delete` checkbox, all sub-checkboxes get disabled, so you have to navigate to the top-level checkbox to disable the scopes again.
It would probably better to always enable the `delete` checkbox so that you can immediately undo an accidental click.

Furthermore, if you click on the `write` checkbox, you enable the `read` checkbox as well.
But if you deselect `write` again,  `read` is still selected.
It would probably be better to be consistent, and deselect the `read` checkbox in this case as well.

These behavior changes don't block the PR for now, but should be fixed in subsequent PRs.